eng/scripts/Verify-And-Merge-PRs.ps1

```diff
@@ -1,111 +1,132 @@
 param(
-  $PRDataArtifactPath,
-  $AuthToken,
-  $ShouldMerge
+  [string]$PRDataArtifactPath,
+  # @('Azure;azure-sdk-for-net;1', ...)
+  [array]$PRDataInline,
+  [string]$AuthToken,
+  [switch]$SkipMerge
 )
 
-. "${PSScriptRoot}\..\common\scripts\logging.ps1"
+. "${PSScriptRoot}/../common/scripts/common.ps1"
 
-$ReadyForMerge = $true
-$mergablePRs = @()
-$headers = @{ "Content-Type" = "text/json" }
-$RetryCount = 5
+$ResolveReviewAuthors = @('copilot-pull-request-reviewer')
 
-if ($null -eq $ShouldMerge) {
-  $ShouldMerge = $null -ne $AuthToken;
-}
+function Main([string]$prFile, [array]$prs, [string]$ghToken, [switch]$noMerge) {
+  # Setup GH_TOKEN for the gh cli commands
+  if ($ghToken) {
+    $env:GH_TOKEN = $ghToken
+  }
 
-if ($AuthToken) {
-  $headers += @{
-    Authorization = "bearer $AuthToken"
+  $prList = @()
+  if ($prFile) {
+    foreach ($line in (Get-Content $prFile)) {
+      $repoOwner, $repoName, $Number = $line.Split(";")
+      $prList += @{ RepoOwner = $repoOwner; RepoName = $repoName; Number = $Number }
+    }
+  }
+  foreach ($line in $prs) {
+    $repoOwner, $repoName, $Number = $line.Split(";")
+    $prList += @{ RepoOwner = $repoOwner; RepoName = $repoName; Number = $Number }
   }
-}
 
-$PRData = Get-Content $PRDataArtifactPath
-# Confirm Mergability
+  $mergeable = ProcessPRMergeStatuses $prList
 
-do 
-{
-  $unMergablePRs = @()
-  foreach ($prDataLine in $PRData)
-  {
-    $repoOwner, $repoName, $prNumber = $prDataLine.Split(";")
+  if ($noMerge) {
+    LogInfo "Skipping merge of $($mergeable.Length) PRs."
+    return
+  }
 
-    $prApiUrl = "https://api.github.com/repos/${repoOwner}/${repoName}/pulls/${prNumber}"
-    $prUrl = "https://github.com/${repoOwner}/${repoName}/pull/${prNumber}"
-    
-    try
+  MergePRs $mergeable
+}
+
+function ProcessPRMergeStatuses([array]$prData) {
+  for ($retry = 1; $retry -le 5; $retry++)
+  {
+    $curr, $prData = $prData, @()
```

Comment:
Is this a fancy way to pop the front of the list? I assume you are only doing it this way because you are modifying the list in the loop.

Reply:
With that said I don' understand the `,@()` part of this expression.

Reply:
It's destructuring the list items into variable assignments, so equivalent to:

```
$curr = $prData
$prData = @()
```

Reply:
I see you are setting prdata to be empty I though you were trying to pull the first item from the prdata array into curr which the name kind of implied.

Reply:
I was trying to think of a better var name, any ideas?

Reply:
Perhaps something like `$currPrData` or `$currPRSet`. I would probably also just split the assignment up and reset $prData on its own line.